p4-16/spec/P4-16-spec.mdk

```diff
@@ -6620,6 +6620,33 @@ table is an abstract finite map that is given a key and produces as a
 result either an action or a "miss" indication, as described in
 Section [#sec-mau-semantics].
 
+The expected meaning of these values is as follows:
+
+- an `exact` match kind on a key field means that the value of the
+  field in the table specifies exactly all the bits for the key field.
+  This is applicable for key fields of all types supported by the target
+  which have an equality operation.
+
+- a `ternary` match kind on a key field means that the field in the
+  table specifies a set of values for the key field using a priority
+  and a mask.  A priority is a numeric value which is used to break
+  ties when a particular key belongs to multiple sets.  When table
+  entries are specified in the P4 program the priorities are generated
+  by the compiler; when entries are specified by the control-plane,
+  the priority may need to be explicitly specified.  Entries with
+  higher priority are matched first.  This specification does not
+  mandate whether "higher" priorities are represented by higher or
+  lower numeric values; this choice is left to the target
+  implementation.
```

Comment:
"this choice is left to the target implementation".  It seems more precise to say that "this choice is determined by the run time API used"